please

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1117
+
+GET_ATTRS=./scripts/get-attrs.nix
+BASH_COMPL_SCRIPT=./scripts/please.complete.bash.sh
+
+
+log() {
+    local args="$*"
+    local PLEASE="\e[32m[please]:\e[0m"
+    echo -e "$PLEASE $args"
+}
+
+log-error() {
+    local args="$*"
+    local PLEASE="\e[31m[please]\e[0m"
+    echo -e "$PLEASE $args"
+}
+
+log-run() {
+    local cmd="$1"
+    log "Running \"$cmd\"\n"
+    eval $cmd
+}
+
+
+_list_attrs() {
+    nix-instantiate --strict --eval --expr "(import $GET_ATTRS {}).all" | tr -d "[]\""
+}
+
+_force_nixpkgs() {
+    # without evaluating this all our queries will fail
+    nix-build --quiet ./nixpkgs-fetch.nix >/dev/null
+}
+
+
+#
+# sanity check functions
+#
+
+_isChannelInstalled() {
+    nix-channel --list | grep contrail >/dev/null
+}
+
+_isSubstituterConfigured() {
+    nix show-config | grep "cache.nix.corp.cloudwatt.com" >/dev/null
+}
+
+_isNixInstalled() {
+    nix --version >/dev/null 2>&1
+}
+
+_sourceNix() {
+    NIX_SH="$HOME/.nix-profile/etc/profile.d/nix.sh"
+    test -f "$NIX_SH" && source "$NIX_SH"
+}
+
+_addCacheConfig() {
+    if test -f ~/.config/nix/nix.conf
+    then
+        log "$HOME/.config/nix/nix.conf exists. Please follow the instructions from the README"
```

Comment:
Wouyld be nice to provide the README section

Reply:
Oh, Yes. Especially now that I realized that I have actually removed that section from the README so I have to fix this. Thanks for noticing.